gensim/models/phrases.py

```diff
@@ -482,19 +479,17 @@ def learn_vocab(sentences, max_vocab_size, delimiter=b'_', progress_per=10000,
                     "PROGRESS: at sentence #%i, processed %i words and %i word types",
                     sentence_no, total_words, len(vocab),
                 )
-            s = [utils.any2utf8(w) for w in sentence]
-            last_uncommon = None
-            in_between = []
-            for word in s:
+            common = []
+            for word in sentence:
+                word = utils.any2utf8(word)
                 if word not in common_terms:
                     vocab[word] += 1
-                    if last_uncommon is not None:
-                        components = it.chain([last_uncommon], in_between, [word])
-                        vocab[delimiter.join(components)] += 1
-                    last_uncommon = word
-                    in_between = []
-                elif last_uncommon is not None:
-                    in_between.append(word)
+                    if common:
+                        common.append(word)
+                        vocab[delimiter.join(common)] += 1
+                    common = [word]
+                elif common:
+                    common.append(word)
                 total_words += 1
```

Comment:
We don't need to increment this every single time, do we?

```suggestion
            total_words += len(sentence)
```